### PR TITLE
fix(streaming): inappropriate serialization for `JoinRow`

### PR DESCRIPTION
**Pull request by yuhao-su**
## What's changed and what's your intention?

- The previous serialization does not match the deserialization, causing bugs.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

### CODE REVIEW

File: src/stream/src/executor/managed_state/join/mod.rs [hunk 1]
```diff
@@ -75,11 +74,12 @@ impl JoinRow {
...
             vec.extend_from_slice(&serialize_cell(v)?)
         }
-        let mut serializer = memcomparable::Serializer::new(vec![]);
 
         // Serialize degree.
-        self.degree.serialize(&mut serializer)?;
-        vec.extend_from_slice(&serializer.into_inner());
+        let mut degree_buf: Vec<u8> = vec![];
+        degree_buf.put_u64_le(self.degree);
+        vec.extend_from_slice(&degree_buf);
```
Comment (skyzh) [inline src/stream/src/executor/managed_state/join/mod.rs:81]: `vec.extend_from_slice(&self.degree.to_le_bytes())` should be enough?

Comment (BowenXiao1999) [reply src/stream/src/executor/managed_state/join/mod.rs:81]: ditto

**Review by BowenXiao1999 — APPROVED** [review]
Rest LGTM

File: src/stream/src/executor/managed_state/join/mod.rs (hunk 1 above)
Comment (BowenXiao1999) [reply src/stream/src/executor/managed_state/join/mod.rs:81]: Emmm didnt notice the auto merge

Comment (yuhao-su) [reply src/stream/src/executor/managed_state/join/mod.rs:81]: Ah sorry, will fix those later.

**Pull request merged**